src/main/java/carpet/CarpetSettings.java

```diff
@@ -274,9 +275,18 @@ private static void set_defaults()
   //                              .extraInfo("Turning nether RNG manipulation on or off."),
   /////rule("endRNG",                "creative", "Turning end RNG manipulation on or off.")
   //                              .extraInfo("Turning end RNG manipulation on or off."),
-  //!rule("viewDistance",          "creative", "Changes the view distance of the server.")
-  //                              .extraInfo("Set to 0 to not override the value in server settings.")
-  //                              .choices("0", "0 12 16 32 64").setNotStrict(),
+  rule("viewDistance",          "creative", "Changes the view distance of the server.")
+                                .extraInfo("Set to 0 to not override the value in server settings.")
+                                .choices("0", "0 12 16 32 64").setNotStrict()
+                                .validate( (s) -> {
+                                      int viewDistance = getInt("viewDistance");
+                                      if (viewDistance < 2)
+                                          viewDistance = ((DedicatedServer) CarpetServer.minecraft_server).getIntProperty("view-distance", 10);
```

Comment:
This will crash on singleplayer, and since carpet now (secondarily) supports singleplayer, you need a check here to see if the minecraft_server is actually a DedicatedServer.

Reply:
Now that you mention it well yeah -.- sure I completely ignored the singleplayer part :-/
Shall the command work in single player too?

Reply:
Of course it should